src/squlearn/encoding_circuit/circuit_library/random_encoding_circuit.py

```diff
@@ -0,0 +1,393 @@
+import numpy as np
+from typing import Union
+import random
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+
+from qiskit.circuit.library import (
+    XGate,
+    YGate,
+    ZGate,
+    HGate,
+    SGate,
+    TGate,
+    CXGate,
+    CYGate,
+    CZGate,
+    CHGate,
+    SwapGate,
+)
+from qiskit.circuit.library import (
+    RXGate,
+    RYGate,
+    RZGate,
+    PhaseGate,
+    CPhaseGate,
+    CRXGate,
+    CRYGate,
+    CRZGate,
+)
+from qiskit.circuit.library import RXXGate, RYYGate, RZZGate, RZXGate
+
+from ..encoding_circuit_base import EncodingCircuitBase
+
+default_gate_weights = {
+    "x": 0.5,
+    "y": 0.5,
+    "z": 0.5,
+    "h": 0.5,
+    "cx": 0.5,
+    "cy": 0.5,
+    "cz": 0.5,
+    "ch": 0.5,
+    "rx": 1.0,
+    "ry": 1.0,
+    "rz": 1.0,
+    "crx": 0.75,
+    "cry": 0.75,
+    "crz": 0.75,
+    "rxx": 0.75,
+    "ryy": 0.75,
+    "rzz": 0.75,
+    "rzx": 0.75,
+}
+
+default_encoding_weights = {
+    "p": 0.5,
+    "x": 0.5,
+    "p_times_x": 0.5,
+    "pi_times_x": 0.5,
+    "arctan_x": 0.5,
+    "p_times_arctan_x": 0.5,
+    "arccos_x": 0.0,
+    "p_times_arccos_x": 0.0,
+}
+
+
+class RandomEncodingCircuit(EncodingCircuitBase):
+    """
+    Random parameterized encoding circuit with randomly picked gates, qubits and feature encodings.
+
+    The random ciruit generation picks gates from a large set of gates
+    (both parameterized and non-parameterized) and places them on randomly drawn qubits.
+    Parameterized gates can have different randomly picked encodings for the features and
+    parameters.
+
+    The probabilites for picking certain gates and encodings can be adjusted with the
+    ``gate_weights`` and ``encoding_weights`` dictionaries.
```

Comment:
Reframed the scentence to "The weight for picking ..." 
All this random routines have the input named "weights" for drawing values from the propability distribution.
Since normalization is not required for the weights, I've kept weights.